utils.lua

```diff
@@ -117,21 +117,13 @@ function utils.computeRanks(scores, gtPos)
 
     if gtPos then
         gtPos = gtPos:view(-1)
-        numOpts = 100
-        ranks = ranks:view(-1, numOpts)
         local gtRanks = torch.LongTensor(gtPos:size(1))
         for i = 1, gtPos:size(1) do
-            gtBinary = torch.LongTensor(numOpts):zero()
-            gtBinary[{gtPos[{i}]}] = 1
-            sorted, rankedIdx = ranks[{i}]:sort()
-            sortedGt = gtBinary:index(1, rankedIdx:long())
-            gtRank = sortedGt:nonzero()
-            gtRanks[i] = gtRank
+            gtRanks[i] = ranks[{i, gtPos[i]}]
```

Comment:
Oh, I see how this works... I tried advanced indexing to avoid this loop (using multiple LongTensors) and when it raised errors about lack of support, it suffered over-simplification (straightaway as the EvalAI script does it).

I tested this to cross verify and it works fine too. We are good to go with it!